apps/api/src/app/rate-limiting/guards/throttler.guard.ts

```diff
@@ -153,6 +153,30 @@ export class ApiRateLimitInterceptor extends ThrottlerGuard implements NestInter
       apiServiceLevel,
     };
 
+    /**
+     * The purpose of the dry run is to allow us to observe how
+     * the rate limiting would behave without actually enforcing it.
+     */
+    const isDryRun = await this.getFeatureFlag.execute(
+      GetFeatureFlagCommand.create({
+        environmentId,
+        organizationId,
+        userId: _id,
+        key: FeatureFlagsKeysEnum.IS_API_RATE_LIMITING_DRY_RUN_ENABLED,
+      })
+    );
+
+    if (isDryRun) {
+      if (!success) {
+        Logger.warn(
+          `[Dry run] ${THROTTLED_EXCEPTION_MESSAGE} orgId: ${organizationId}, envId: ${environmentId}`,
+          'ApiRateLimitInterceptor'
+        );
+      }
+
+      return true;
+    }
+
```

Comment:
I think we should move this above the `res.header` setters to avoid any confusion of headers being sent in the response without being acted upon.

Otherwise, looks great!

Reply:
Done - also added a bit more info to the log message for potential debugging purpose , since the headers won't be send now